azerothcore/azerothcore-wotlk
# fix(setup): MySQL 5.7

##### CHANGES PROPOSED:

-  Fix MySQL 5.7 setup issue
-  No longer need to change sql mode


###### ISSUES ADDRESSED:


Updates https://github.com/azerothcore/azerothcore-wotlk/issues/1134

##### TESTS PERFORMED:


- Tested on Ubuntu with MySQL 5.7 with a **clean install** (no existing databases)
- Also tested on Ubuntu with Docker


##### HOW TO TEST THE CHANGES:

On a traditional setup:
- Make sure your MySQL 5.7 server is running with the default sql mode
- Delete all your AC DBs
- Install AC from scratch

On docker:
- Stop all your containers (`docker-compose down`)
- Clean all containers and images (`docker system prune -a` **THIS WILL DELETE ALL IMAGES/CONTAINERS IN YOUR PC**)
- Install AC with Docker again from scratch


##### KNOWN ISSUES:

Not sure why Travis still needs to reset the sql mode otherwise fails with

> ERROR 1133 (42000) at line 1: Can't find any matching row in the user table

Probably because of default `NO_AUTO_CREATE_USER`.

See: https://stackoverflow.com/questions/17597144/cant-find-any-matching-row-in-the-user-table

##### Target branch(es):

Master
